### DISCUSSION TITLE
How to call the scan function G38.2 in GrblHAL

### DISCUSSION THREAD

**Discussion by GrblGru** [grblHAL/core#322] [Q&A] [answered]
Sorry, I'm a little confused right now. Therefore maybe my stupid question.

I just wanted to test the function G38.2 with GrblHAL and I get the error message 20 "Unsupported or invalid g-code command found in block". My search for "g38.2" on this forum was unsuccessful.

Can someone tell me what is wrong with the command "G91 g38.2 z-10.000 f100" ?
GrblHAL provides this important function ... or ?

**Comment by terjeio** [comment]
If your driver/board combo does not support a probe input the G38.x is not available. You may use the `$pins` command to check.
FYI in the Web Builder probe input is a selectable option.

**Reply by GrblGru** [reply]
Thank you for your quick reply.  :)
I still do not understand what I have to select under "board" if I have for example a normal Arduino Due and connected it via cable with external drivers. So in my understanding I don't have a board in the real sense.  I have therefore selected Ramps V1.6. The option "Probe Input" I had also selected.

**Reply by terjeio** [reply]
That is odd, could be a Web Builder issue. The `$pins` command has no info about the probe input?

**Reply by GrblGru** [reply]
I can't find one.
But I will flash the due again to be sure. Is it correct if I select "Ramps 1.6" as board ? 

![grafik](https://github.com/grblHAL/core/assets/60187049/f29d55dd-d7b7-4dd0-ae59-4e6e4109a644)

**Reply by terjeio** [reply]
Hmm, I'll have to check this out as the map file has the probe pin defined:
```
// Define probe switch input pin.
#define PROBE_PORT         	PIOB
#define PROBE_PIN        	13  // Due Digital Pin 21
```

**Reply by GrblGru** [reply]
Sorry, my mistake. Just flashed the due again. Probe' is now available. G38.2 does not give the error now either. Must still wire everything. Will try everything tonight.

Thanks a lot for your great help. :)

![grafik](https://github.com/grblHAL/core/assets/60187049/402048eb-cf25-4af0-9d4c-9089a91a545e)

**Reply by GrblGru** [reply]
Sorry that I have to bother you again.
I now have a STM32 blue pill here and am testing the probing.
It works also so far well. When I look at PRB with $# after the measurement it looks good.

But I don't get a "[PRB: ..." message at runtime. I suppose that I can set it somewhere, but I can't find it.

**Reply by terjeio** [reply, marked as answer]
It is enabled by the extended `$10` setting flags (on by default when $10 = 511).
If you send `$$=10` you get this output:
```
    0 - Position in machine coordinate (1)
    1 - Buffer state (2)
    2 - Line numbers (4)
    3 - Feed & speed (8)
    4 - Pin state (16)
    5 - Work coordinate offset (32)
    6 - Overrides (64)
    7 - Probe coordinates (128)
    8 - Buffer sync on WCO change (256)
    9 - Parser state (512)
    10 - Alarm substatus (1024)
    11 - Run substatus (2048)
```

**Reply by GrblGru** [reply]
Thanks again. You are the best.  :)
